site/tests/integration/blockpage.test.ts

```diff
@@ -0,0 +1,113 @@
+import { expect, test } from "playwright-test-coverage";
+
+import blocksData from "../../blocks-data.json";
+
+const codeBlock = blocksData.find(
+  ({ name }) => name === "@hashintel/block-code",
+);
+
+const unsupportedBlock = blocksData.find(
+  ({ name }) => name === "@hashintel/block-embed",
+);
+
+if (!codeBlock || !unsupportedBlock) {
+  throw new Error("Code and Embed blocks need to be prepared before tests run");
+}
+
+test("Block page should contain key elements", async ({ page, isMobile }) => {
+  await page.goto(codeBlock.blockPackagePath);
+
+  await expect(
+    page.locator(`h1:has-text('${codeBlock.displayName}')`),
+  ).toBeVisible();
+
+  await expect(
+    page.locator(`p:has-text('${codeBlock.description}')`),
```

Comment:
Generally speaking, we should escape `'` in values here. I _guess_ we can get away with the selectors as is because `codeBlock` is owned by HASH so we should not anticipate query injections from ourselves 😅 The syntax will be broken though is someone adds `'` to description.